src/client/connection.ts

```diff
@@ -58,6 +58,22 @@ function getAgent(url: string, config: ConnectionOptions): Agent | undefined {
   if (config.proxy != null) {
     const parsedURL = new URL(url)
     const parsedProxyURL = new URL(config.proxy)
+    const proxyInfo = _.omitBy(
+      {
+        href: parsedProxyURL.href,
+        origin: parsedProxyURL.origin,
+        protocol: parsedProxyURL.protocol,
+        username: parsedProxyURL.username,
+        password: parsedProxyURL.password,
+        host: parsedProxyURL.host,
+        hostname: parsedProxyURL.hostname,
+        port: parsedProxyURL.port,
+        pathname: parsedProxyURL.pathname,
+        search: parsedProxyURL.search,
+        hash: parsedProxyURL.hash,
+      },
+      (value) => value == null,
```

Comment:
Do we even need to do this `omitBy`? Isn't JS going to handle that on its own?

Reply:
Yeah, you're prolly right that we can just get rid of it.